openmc/data/angle_distribution.py

```diff
@@ -32,6 +32,16 @@ def __init__(self, energy, mu):
         self.energy = energy
         self.mu = mu
 
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        else:
+            eqval = True
+            if (not np.array_equal(self.energy, other.energy) or
+                not np.array_equal(self.mu, other.mu)):
```

Comment:
Hmm, just thinking about what will happen when I extend our capabilities to [read ENDF files](https://github.com/paulromano/openmc/tree/pyapi-data-endf) and then mu might be a mix of tabulated and Legendre distributions. Not sure if there's a good way to handle that.

ENDF issues aside -- would this really work with mu being a list of `Tabular` objects?